.github/workflows/pre-commit-autoupdate.yml

```diff
@@ -0,0 +1,79 @@
+name: "Run pre-commit autoupdate"
+
+on:
+  workflow_call:
+    inputs:
+      target:
+        description: "The branch to issue the PR against"
+        type: "string"
+        default: "main"
+    secrets:
+      TOKEN:
+        description: "Your token"
+        required: true
+
+permissions:
+  contents: write
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  precommit-autoupdate:
+    name: "pre-commit autoupdate"
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Check out repository"
+        uses: actions/checkout@v4
+
+      - name: "Create a new branch"
+        id: branch
+        run: |
+          branch=pre-commit/$(date +'%Y-%m-%dT%H_%M_%S')
+          git checkout -b $branch
+          echo "name=$branch" >> $GITHUB_OUTPUT
+
+      - name: "Add brew to the PATH"
+        run: echo "/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin" >> $GITHUB_PATH
+
+      - name: "Install pre-commit"
+        run: brew install pre-commit
+
+      - name: "Check for pre-commit hook updates"
+        id: pre-commit
+        run: |
+          pre-commit autoupdate
+          updates=$(git diff --raw)
+          echo "updates=$updates" >> $GITHUB_OUTPUT
+
+      - name: "Commit local changes"
+        if: ${{ steps.pre-commit.outputs.updates }}
+        run: |
+          git config user.name "$USER"
+          git config user.email "$EMAIL"
+          git status
+          git add .
+          git commit -m "$MESSAGE"
+        env:
+          USER: "Github Build Bot"
+          EMAIL: "buildbot@fishtownanalytics.com"
+          MESSAGE: "[automated] update pre-commit hooks"
+
+      - name: "Open pull request"
+        id: pull_request
+        run: |
+          git push --set-upstream origin $HEAD
+          url=$(gh pr create -B $TARGET -H $HEAD -t "$TITLE" -b "$BODY")
```

Comment:
It makes it easier for the next person if you use the full qualifications (`--title` instead of `-t`, etc) esp if they're less familiar with the gh cli

Reply:
Agreed, I'll update.